components/global/core/io.cellery.observability.api/src/main/java/io/cellery/observability/api/internal/RegisterClient.java

```diff
@@ -0,0 +1,130 @@
+package io.cellery.observability.api.internal;
+
+import io.cellery.observability.api.AggregatedRequestsAPI;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.config.Registry;
+import org.apache.http.config.RegistryBuilder;
+import org.apache.http.conn.HttpClientConnectionManager;
+import org.apache.http.conn.socket.ConnectionSocketFactory;
+import org.apache.http.conn.socket.PlainConnectionSocketFactory;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.conn.BasicHttpClientConnectionManager;
+import org.apache.log4j.Logger;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+/**
+ * This class is used to create a DCR request for Service Provider registeration.
+ */
+public class RegisterClient {
+
+    private static final Logger log = Logger.getLogger(AggregatedRequestsAPI.class);
+
+    public static JSONObject getClientCredentials() throws IOException {
+        BufferedReader bufReader = null;
+        JSONObject obj2 = null;
+        try {
+
+            HttpClient client = getAllSSLClient();
+
+            HttpPost request =
+                    new HttpPost("https://gateway.cellery-system:9443/client-registration/v0.14/register");
+            JSONObject obj = new JSONObject();
+            obj.put("callbackUrl", "http://cellery-dashboard");
+            obj.put("clientName", "Cellery-Observability-Portal");
+            obj.put("owner", "admin");
+            obj.put("grantType", "authorization_code");
+            obj.put("saasApp", " true");
+            request.setHeader("Authorization", "Basic YWRtaW46YWRtaW4=");
+            request.setHeader("content-type", "application/json");
+            request.setEntity(new StringEntity(obj.toString()));
+
+            HttpResponse response = client.execute(request);
+
+            bufReader = new BufferedReader(new InputStreamReader(
+                    response.getEntity().getContent(), StandardCharsets.UTF_8));
+
+            StringBuilder builder = new StringBuilder();
+
+            String line;
+
+            while ((line = bufReader.readLine()) != null) {
+                builder.append(line);
+                builder.append(System.lineSeparator());
+            }
+
+            obj2 = new JSONObject(builder.toString());
+            return obj2;
+
+        } catch (KeyStoreException | NoSuchAlgorithmException | KeyManagementException e) {
+            log.error("Message", e);
```

Comment:
Always add a meaningful error message. For an example "Error while retrieving client credentials. "